packages/graphback-runtime-knex/src/KnexDBDataProvider.ts

```diff
@@ -39,26 +39,21 @@ export class KnexDBDataProvider<Type = any> implements GraphbackDataProvider<Typ
     const { idField, data: updateData } = getDatabaseArguments(this.tableMap, data);
 
     //tslint:disable-next-line: await-promise
-    const updateResult = await this.db(this.tableName).update(updateData).where(idField.name, '=', idField.value);
-    if (updateResult === 1) {
-      //tslint:disable-next-line: await-promise
-      const dbResult = await this.db.select(this.getSelectedFields(selectedFields)).from(this.tableName).where(idField.name, '=', idField.value);
-      if (dbResult && dbResult[0]) {
-        return dbResult[0]
-      }
+    const updateResult = await this.db(this.tableName).update(updateData, this.getSelectedFields(selectedFields)).where(idField.name, '=', idField.value);
+    if (updateResult && updateResult[0]) {
+      return updateResult[0];
     }
```

Comment:
@wtrocki I'd love to know the original reason for two separate queries in the first place - I recall there being a reason so don't want to approve this without input on this

Reply:
Support for MYSQL, Postgres. Some dbs return different data. For PG this is good

Reply:
Thanks for the context. But what this patch brings is no different from https://github.com/aerogear/graphback/blob/9615ffb50453b9772b39eda22d645f16dd91569a/packages/graphback-runtime-knex/src/KnexDBDataProvider.ts#L31

To have a more general purpose Relational database data provider, eventually, I'd see us renaming the  [SQLiteKnexDBDataProvider](https://github.com/aerogear/graphback/blob/9615ffb50453b9772b39eda22d645f16dd91569a/packages/graphback-runtime-knex/src/SQLiteKnexDBDataProvider.ts) to something more general and guide people on how to integrate it in their application. Much like what is done here https://graphback.dev/docs/databases/sqlite#using-knex-provider

Reply:
Yep. I did not meant to say your change is somehow wrong. We should merge it. eventually check docs for sqlite